src/entity-manager/EntityManager.ts

```diff
@@ -53,7 +53,7 @@ export class EntityManager {
     /**
      * Connection used by this entity manager.
      */
-    readonly connection: DataSource
+    readonly connection: Omit<DataSource, "manager">
```

Comment:
Proposed solution doesn't look appropriate. Why we need to omit manager property from the DataSource type?

Reply:
Maybe to avoid some circular dependency.

Reply:
no, circular dependencies don't get resolved in a such way. Need to fix source of circular and re-structure the code, or try to put `import type` statements for circular deps issue.